text/0000-app-mixins.md

```diff
@@ -0,0 +1,241 @@
+* If the mixin name is prefixed with `run:` (as per [RFC-0006](https://github.com/buildpacks/rfcs/blob/main/text/0006-stage-specific-mixins.md)) the mixin will be dynamically added to the run image.
+* If the mixin name does not have a prefix it will be dynamically added to both the build and run stack images.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+When a list of mixins are defined in a `project.toml` and `pack build` is run against that project, the following will happen:
+
+1. The lifecycle will compare the list of mixins to those provided by the stack. If all mixin names are provided by the stack, no further action is required.
+1. In any requested mixin is not provided by the stack, the lifecycle will run the detect phase for all stackpacks defined in the builder.
+1. The lifecycle will compare the mixins added to the build plan to see if they match they required mixins.
+1. If at least one stackpack passes detect and provides the required mixin(s), the lifecycle will execute the stackpack build phase for passing stackpack(s). If no stackpacks pass detect, or no stackpacks provide the required mixin, the build will fail with a message that describes the mixins that could not be provided.
+1. During the lifecycle's build phase, the stackpacks that passed detection will run against the build and run images accordingly (see details below). All stackpacks will be run before the user's buildpacks.
+
+## Stackpacks
+
+ A stackpacks is a special case of buildpack that has the following properties:
+
+* Is run as the `root` user
+* Configured with `privileged = true` in the `buildpack.toml`
+* Can only create one layer
+* Does not have access to the application source code
+* Is run before all regular buildpacks
+* Is run against both the build and run images
+* Is distributed with the builder image
+* May not write to the `/layers` directory
+
+The stackpack interface is identical to the buildpack interface (i.e. the same `bin/detect` and `bin/build` scripts are required). However, some of the context it is run in is different from regular buildpacks.
+
+For each stackpack, the lifecycle will use [snapshotting](https://github.com/GoogleContainerTools/kaniko/blob/master/docs/designdoc.md#snapshotting-snapshotting) to capture changes made during the stackpack's build phase (excluding `/tmp`, `/cnb`, and `/layers`). Alternatively, a platform may store a new stack image to cache the changes. All of the captured changes will be included in a single layer produced as output from the stackpack. The `/layers` dir MAY NOT be used to create arbitrary layers.
+
+A stackpack is included in a builder by defining it in the `builder.toml` in the `[[stack.buildpacks]]` array of tables:
+
+```
+[[stack.buildpacks]]
+id = "<stackpack id>"
+uri = "<uri to stackpack>"
+```
+
+Each stackpack included in the builder will execute until all of a project's mixins have been provided.
```

Comment:
This wording to me reads as ~"once a project's mixins have been provided by an earlier stackpack in the execution order, remaining stackpacks won't be executed".

However in the "how it works" section, the wording is:

> If at least one stackpack passes detect and provides the required mixin(s), the lifecycle will execute the stackpack build phase for passing stackpack(s).

...which suggests that instead there is no "early return" and all passing stackpacks will be run, even if an earlier one in the sequence provides the required mixin?